test/testdata/lsp/completion/missing_assign_rhs.rb

```diff
@@ -12,4 +12,15 @@ def test_variable_after_method(x)
     y =
     #  ^ completion: x, y, foo, test_missing_rhs, ...
   end # error: unexpected token "end"
+
+  def test_variable_end_same_line(x)
+    puts 'before'
+    y =end # error: unexpected token "end"
+    #  ^ completion: x, y, foo, test_missing_rhs, ...
+
+  def test_variable_end_same_line_after(x)
+    # This one is admittedly janky. There's a note about it in the parser.
+    puts 'before'
+    y =end # error: unexpected token "end"
+    #     ^ completion: x, y, foo, test_missing_rhs, ...
```

Comment:
So I read the comment in the parser that you're referencing, but still having trouble understanding why the completion pops up in a different location for `test_variable_end_same_line_after`, whose body is lexically identical to `test_variable_end_same_line` (except for the comment... maybe that matters?). I guess the error tokens in these two examples must span different locations, but I don't understand why, and I can't work out exactly which ranges they're spanning. Could you elaborate a bit?

Reply:
LSP queries never match on AST nodes that have zero width.

If we had used `@2.end, @3.begin` when creating the `error_node`, that would be the end of the `=` and the start of the `end`, which would thus give that AST node zero width, so that `error_node` would never participate in LSP query responses.

Instead, the LSP query would return the next-enclosing AST nodde, which in that case would **maybe** be the assignment node `x =`, but in fact I think LSP properly treats end positions in locs as exclusive, so it would actually result in a query for the enclosing method def. Or something idk. It for sure wouldn't be the error_node.